scene/3d/visual_instance_3d.cpp

```diff
@@ -53,13 +86,46 @@ void VisualInstance3D::_notification(int p_what) {
 		} break;
 
 		case NOTIFICATION_TRANSFORM_CHANGED: {
-			Transform3D gt = get_global_transform();
-			RenderingServer::get_singleton()->instance_set_transform(instance, gt);
+			if (_is_vi_visible() || is_physics_interpolated_and_enabled()) {
+				if (!_is_using_identity_transform()) {
+					Transform3D gt = get_global_transform();
+					RenderingServer::get_singleton()->instance_set_transform(instance, gt);
+
+					// For instance when first adding to the tree, when the previous transform is
+					// unset, to prevent streaking from the origin.
+					if (_is_physics_interpolation_reset_requested()) {
+						if (_is_vi_visible()) {
+							_notification(NOTIFICATION_RESET_PHYSICS_INTERPOLATION);
+						}
+						_set_physics_interpolation_reset_requested(false);
+					}
+				}
```

Comment:
```suggestion
			if ((_is_vi_visible() || is_physics_interpolated_and_enabled()) && !_is_using_identity_transform()) {
				Transform3D gt = get_global_transform();
				RenderingServer::get_singleton()->instance_set_transform(instance, gt);

				// For instance when first adding to the tree, when the previous transform is
				// unset, to prevent streaking from the origin.
				if (_is_physics_interpolation_reset_requested()) {
					if (_is_vi_visible()) {
						_notification(NOTIFICATION_RESET_PHYSICS_INTERPOLATION);
					}
					_set_physics_interpolation_reset_requested(false);
				}
```